query/compile.go

```diff
@@ -764,6 +768,19 @@ func (c *compiledStatement) subquery(stmt *influxql.SelectStatement) error {
 		return err
 	}
 
+	// If the ordering is different for the subquery than the parent, throw an
+	// error. Ordering can only be specified at the top level.
+	if subquery.Ascending != c.Ascending {
+		// If the ordering is different, but the subquery has no sort fields
+		// (so the user never specified they wanted it to be ascending), inherit
+		// the ordering from the parent query.
+		if len(stmt.SortFields) == 0 {
+			subquery.Ascending = c.Ascending
+		} else {
+			return errors.New("subqueries must be ordered in the same direction as the query itself")
+		}
+	}
```

Comment:
Do the sort fields need to be the same if `Ascending` is the same?

Reply:
Well, the sort fields have to be the same if `Ascending` is the same. Although I think we could largely change this logic to `if len(stmt.SortFields) != 0 && subquery.Ascending != c.Ascending`. I'll change it to that so there's less nesting.